packages/netlify-cms-core/src/actions/mediaLibrary.js

```diff
@@ -186,14 +189,39 @@ export function persistMedia(file, opts = {}) {
       const id = await getBlobSHA(file);
       const assetProxy = await createAssetProxy(fileName, file, false, privateUpload);
       dispatch(addAsset(assetProxy));
+
+      const entry = state.entryDraft.get('entry');
+      const useWorkflow = state.config.getIn(['publish_mode']) === EDITORIAL_WORKFLOW;
+      const draft = entry && !entry.isEmpty() && useWorkflow;
+
       if (!integration) {
-        const asset = await backend.persistMedia(state.config, assetProxy);
+        const asset = await backend.persistMedia(state.config, assetProxy, draft);
```

Comment:
We still need the backend to generate the asset object so we call `persistMedia `